[auth] Add ID state code to SSO users

Catacola opened the pull request:
## Description of the change

Adds a new action to the beginning of the login flow. This action checks for the lack of a state code on a user and populates it based on the user's connection id if missing. The connection ids are stored as secrets in Auth0 to allow for different configs in staging and prod.

## Type of change

- [x] New feature (non-breaking change that adds functionality)

## Related issues

Part of #1429

## Checklists

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

jovergaag reviewed (approved):
Did you test this out in `recidiviz-dev` with a new US_ID user?

Catacola commented:
> Did you test this out in `recidiviz-dev` with a new US_ID user?

I needed to test it in prod due to the SSO integration config on their side. I turned it on, tested, and turned it back off. I discovered a new issue where some users from idaho come up with a `@idoc.onmicrosoft.com` domain. This should fix that.

Catacola commented:
@phenggeler @jovergaag I'm requesting a rereview since I changed how the state code mapping is done and I added in the bit of logic to account for idaho users possibly having a different email domain.
